nucleus/admin/server-mgmt/src/test/java/fish/payara/admin/servermgmt/cli/PrintCertificateCommandTest.java

```diff
@@ -0,0 +1,287 @@
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests the {@link PrintCertificateCommand} with default and BouncyCastle provider.
+ * Verifies work with several file formats.
+ *
+ * @author David Matejcek
+ */
+@RunWith(Parameterized.class)
+public class PrintCertificateCommandTest {
+
+    private static final String ALIAS = "test";
+
+    private static final String KEYSTORE_PASSWORD = "changeit";
+
+    private static final File FILE_JKS = new File("target/pcct.jks");
+    private static final File FILE_PKCS12 = new File("target/pcct.p12");
+    private static final File FILE_JCEKS = new File("target/pcct.jceks");
+    private static final File FILE_DER = new File("target/pcct.der");
+    private static final File FILE_PEM = new File("target/pcct.pem");
+
+    private static final PrintStream ORIGINAL_STDOUT = System.out;
+    private static final String ORIGINAL_INSTALL_ROOT = System.getProperty(INSTALL_ROOT_PROPERTY);
+
+    private ByteArrayOutputStream stdout;
+    private int providerCount;
+
+    private final PrintCertificateCommandMock command;
+
+
+    @BeforeClass
+    public static void initEnvironment() throws Exception {
+        // prevents NPE in CLICommand
+        if (ORIGINAL_INSTALL_ROOT == null) {
+            System.setProperty(INSTALL_ROOT_PROPERTY, ".");
+        }
+
+        final KeyPair keyPair = createKeyPair();
+        final X509Certificate certificate = createSelfSignedCertificate(keyPair);
+
+        saveKeyStore(keyPair.getPrivate(), certificate, FILE_PKCS12, "PKCS12");
+        saveKeyStore(keyPair.getPrivate(), certificate, FILE_JKS, "JKS");
+        saveKeyStore(keyPair.getPrivate(), certificate, FILE_JCEKS, "JCEKS");
+        saveDer(certificate);
+        savePem(certificate);
+    }
+
+
+    @AfterClass
+    public static void resetEnvironment() {
+        if (ORIGINAL_INSTALL_ROOT == null) {
+            System.clearProperty(INSTALL_ROOT_PROPERTY);
+        } else {
+            System.setProperty(INSTALL_ROOT_PROPERTY, ORIGINAL_INSTALL_ROOT);
+        }
+        System.setOut(ORIGINAL_STDOUT);
+    }
+
+
+    @Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] { //
+            {FILE_DER, null}, {FILE_PEM, null}, {FILE_PKCS12, ALIAS}, {FILE_JKS, ALIAS}, {FILE_JCEKS, ALIAS} //
+        });
+    }
+
+
+    public PrintCertificateCommandTest(final File file, final String alias) {
+        this.command = new PrintCertificateCommandMock();
+        this.command.certificateAlias = alias;
+        this.command.file = file.getAbsolutePath();
+    }
+
+
+    @Before
+    public void init() {
+        this.stdout = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(this.stdout));
```

Comment:
I don't think we want to change output stream but not restore it after the test. Maybe that is done by JUnit? Don't know.

Reply:
It is restored via @BeforeClass - @AfterClass mechanism.

Reply:
Funny, I was looking for that and didn't see it then but had no problem now.